sentry_docker_conf.py

```diff
@@ -240,3 +241,27 @@ def nydus_config(from_env_var):
     logger.addHandler(logging.StreamHandler())
     ldap_loglevel = getattr(logging, config('LDAP_LOGLEVEL', default='DEBUG'))
     logger.setLevel(ldap_loglevel)
+
+##############################
+# REMOTE_USER authentication #
+##############################
+
+SENTRY_USE_REMOTE_USER = config('SENTRY_USE_REMOTE_USER', default=False, cast=bool)
+
+if SENTRY_USE_REMOTE_USER:
+    AUTHENTICATION_BACKENDS += ('django.contrib.auth.backends.RemoteUserBackend',)
+
+    AUTH_REMOTE_USER_HEADER = config('AUTH_REMOTE_USER_HEADER', default=None)
+    if AUTH_REMOTE_USER_HEADER:
+        # The lazy hack is required because importing RemoteUserMiddleware at load time leads to a circular import
+        # The name is upper camel case because that's the only way to expose values from this config file
+        def build_LAZY_CUSTOM_REMOTE_USER_MIDDLEWARE(header, *args, **kwargs):
+            from django.contrib.auth.middleware import RemoteUserMiddleware
+            class CustomRemoteUserMiddleware(RemoteUserMiddleware):
+               pass
+            CustomRemoteUserMiddleware.header = header
+            return CustomRemoteUserMiddleware(*args, **kwargs)
+        LAZY_CUSTOM_REMOTE_USER_MIDDLEWARE = functools.partial(build_LAZY_CUSTOM_REMOTE_USER_MIDDLEWARE, AUTH_REMOTE_USER_HEADER)
+        MIDDLEWARE_CLASSES += ('sentry_config.LAZY_CUSTOM_REMOTE_USER_MIDDLEWARE',)
```

Comment:
The Django settings module doesn't have a schema. All the Django config system does is it imports every value from this file that's in `UPPERCASE_UNDERSCORE`. That covers the `LAZY...` part.

As for `sentry_config`, I didn't know either (just saw the module name while trying to implement this). After some digging it turns out that Sentry uses Logan (like this: https://github.com/getsentry/sentry/blob/master/src/sentry/utils/runner.py#L398) to run itself as a stand-alone Django app. And Logan defines the settings module name as `${PROJECT_NAME}_config`: https://github.com/dcramer/logan/blob/master/logan/runner.py#L88